sphinxcontrib/dotnetdomain.py

```diff
@@ -33,7 +33,8 @@
     [\>\}])(?![\<\{][^\>\}]+[\>\}])
 '''
 _re_parts['type'] = r'(?:%(type_dimension)s|%(type_generic)s)' % _re_parts
-_re_parts['name'] = r'[\w\_\-]+?%(type)s' % _re_parts
+_re_parts['type_indexer'] = r'(\[[\w\_\-\.]+?%(type)s\])?' % _re_parts
+_re_parts['name'] = r'[\w\_\-]+?(?:%(type)s|%(type_indexer)s)' % _re_parts
```

Comment:
So hopefully this is mostly explanatory, but just in case:

`_re_parts['name']` should be an alpha numeric string followed by either a type definition (generic, etc) or an indexer syntax, which is very close to the generic syntax anyways.